swift-sdk/Internal/EmbeddedMessagingProcessor.swift

```diff
@@ -5,51 +5,48 @@
 import Foundation
 
 struct EmbeddedMessagingProcessor {
-    init(currentMessages: [IterableEmbeddedMessage], fetchedMessages: [IterableEmbeddedMessage]) {
+    init(currentMessages: [Int: [IterableEmbeddedMessage]], fetchedMessages: [Int: [IterableEmbeddedMessage]]) {
         self.currentMessages = currentMessages
         self.fetchedMessages = fetchedMessages
     }
-    
-    func processedMessagesList() -> [IterableEmbeddedMessage] {
-        // TODO: understand/handle case of message with same message ID but different contents
-        
-        return fetchedMessages
-    }
 
-    func newlyRetrievedMessages() -> [IterableEmbeddedMessage] {
-        return getNewMessages()
-    }
-    
-    func newlyRemovedMessageIds() -> [String] {
-        return getCurrentMessageIds().filter { !getFetchedMessageIds().contains($0) }
+    func processedMessagesList() -> [Int: [IterableEmbeddedMessage]] {
+        return fetchedMessages
     }
 
-//    func placementIdsToNotify() -> [String] {
-//        // TODO: account for removed placement IDs, this only counts new ones for now
-//        
-//        return getNewMessages()
-//            .map { $0.metadata.placementId }
-//    }
-    
-    // MARK: - PRIVATE/INTERNAL
-    
-    private let currentMessages: [IterableEmbeddedMessage]
-    private let fetchedMessages: [IterableEmbeddedMessage]
-    
-    private func getNewMessages() -> [IterableEmbeddedMessage] {
-        let currentMessageIds = currentMessages.map { $0.metadata.messageId }
+    func newlyRetrievedMessages() -> [Int: [IterableEmbeddedMessage]] {
+        var newMessages: [Int: [IterableEmbeddedMessage]] = [:]
```

Comment:
Looks like this is processing embedded messages and storing them in a key pair structure.